packages/scoutgame-ui/src/components/common/NFTListingPurchase/NFTListingPurchaseDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import { RainbowKitProvider, useConnectModal } from '@rainbow-me/rainbowkit';
+import { useEffect, useState } from 'react';
+import { useAccount } from 'wagmi';
+
+import { useSmScreen } from '../../../hooks/useMediaScreens';
+import { Dialog } from '../Dialog';
+
+import type { NFTListingPurchaseFormProps } from './NFTListingPurchaseForm';
+import { NFTListingPurchaseForm } from './NFTListingPurchaseForm';
+
+type NFTListingPurchaseDialogProps = {
+  open: boolean;
+  onClose: VoidFunction;
+  listing: NFTListingPurchaseFormProps['listing'];
+  builder: NFTListingPurchaseFormProps['builder'];
+};
+
+function NFTListingPurchaseDialogComponent({ listing, open, onClose, builder }: NFTListingPurchaseDialogProps) {
```

Comment:
it seems like this component could be generalized as a wrapper, so we dont have it repeated 3 times. its main intent is just to make sure u connect a wallet before u get to the form, right?

not a huge concern at the moment tho

Reply:
lets do it for v2. we have 3 separate places you're right. nft purchase, nft listing and nft listing purchase.